PyDynamic/uncertainty/propagate_MonteCarlo.py

```diff
@@ -565,10 +565,12 @@ def _UMCevaluate(th, nbb, x, Delta, phi, theta, sigma, blow, alow):
     return lfilter(bb, aa, xlow) + d
 
 
-def UMC_generic(draw_samples, evaluate, runs = 100, blocksize = 8, runs_init = 10, nbins = 100, return_samples = False, n_cpu = multiprocessing.cpu_count()):
+def UMC_generic(draw_samples, evaluate, runs = 100, blocksize = 8, runs_init = 10, nbins = 100,
+                return_samples = False, n_cpu = multiprocessing.cpu_count()):
     """
     Generic Batch Monte Carlo using update formulae for mean, variance and (approximated) histogram.
     Assumes that the input and output of evaluate are numeric vectors (but not necessarily of same dimension).
+    If the output of evaluate is multi-dimensional, it will be flattened into 1D. 
 
     Parameters
     ----------
```

Comment:
evaluate:
needs to return a (multi dimensional) numpy.ndarray

draw_samples:
needs to return a list of (multi dimensional) numpy.ndarrays